kornia/enhance/jpeg.py

```diff
@@ -0,0 +1,566 @@
+        output_cb (Tensor): Cb component (subsampled), shape is :math:`(B, H // 2, W // 2)`.
+        output_cr (Tensor): Cr component (subsampled), shape is :math:`(B, H // 2, W // 2)`.
+    """
+    # Get components
+    output_y: Tensor = input_ycbcr[:, 0]
+    output_cb: Tensor = input_ycbcr[:, 1]
+    output_cr: Tensor = input_ycbcr[:, 2]
+    # Perform average pooling of Cb and Cr channels
+    output_cb = rescale(output_cb[:, None], factor=0.5, interpolation="bilinear", align_corners=False, antialias=True)
+    output_cr = rescale(output_cr[:, None], factor=0.5, interpolation="bilinear", align_corners=False, antialias=True)
+    return output_y, output_cb[:, 0], output_cr[:, 0]
+
+
+def _chroma_upsampling(input_c: Tensor) -> Tensor:
+    """Function performs chroma upsampling.
+
+    Args:
+        input_c (Tensor): Cb or Cr component to be upsampled of the shape :math:`(B, H, W)`.
+
+    Returns:
+        output_c (Tensor): Upsampled C(b or r) component of the shape :math:`(B, H * 2, W * 2)`.
+    """
+    # Upsample component
+    output_c: Tensor = rescale(
+        input_c[:, None], factor=2.0, interpolation="bilinear", align_corners=False, antialias=False
+    )
+    return output_c[:, 0]
+
+
+def _jpeg_encode(
+    image_rgb: Tensor,
+    jpeg_quality: Tensor,
+    quantization_table_y: Tensor,
+    quantization_table_c: Tensor,
+) -> Tuple[Tensor, Tensor, Tensor]:
+    """Performs JPEG encoding.
+
+    Args:
+        image_rgb (Tensor): RGB input images of the shape :math:`(B, 3, H, W)`.
+        jpeg_quality (Tensor): Compression strength of the shape :math:`(B)`.
+        quantization_table_y (Tensor): Quantization table for Y channel.
+        quantization_table_c (Tensor): Quantization table for C channels.
+
+    Returns:
+        y_encoded (Tensor): Encoded Y component of the shape :math:`(B, N, 8, 8)`.
+        cb_encoded (Tensor): Encoded Cb component of the shape :math:`(B, N, 8, 8)`.
+        cr_encoded (Tensor): Encoded Cr component of the shape :math:`(B, N, 8, 8)`.
+    """
+    # Convert RGB image to YCbCr.
+    image_ycbcr: Tensor = rgb_to_ycbcr(image_rgb)
+    # Scale pixel-range to [0, 255]
+    image_ycbcr = 255.0 * image_ycbcr
+    # Perform chroma subsampling
+    input_y, input_cb, input_cr = _chroma_subsampling(image_ycbcr)
+    # Patchify, DCT, and rounding
+    input_y, input_cb, input_cr = _patchify_8x8(input_y), _patchify_8x8(input_cb), _patchify_8x8(input_cr)
+    dct_y, dct_cb, dct_cr = _dct_8x8(input_y), _dct_8x8(input_cb), _dct_8x8(input_cr)
+    y_encoded: Tensor = _quantize(
+        dct_y,
+        jpeg_quality,
+        quantization_table_y,
+    )
+    cb_encoded: Tensor = _quantize(
+        dct_cb,
+        jpeg_quality,
+        quantization_table_c,
+    )
+    cr_encoded: Tensor = _quantize(
+        dct_cr,
+        jpeg_quality,
+        quantization_table_c,
+    )
+    return y_encoded, cb_encoded, cr_encoded
+
+
+def _jpeg_decode(
```

Comment:
would be interesting to include this one in the high level api as well (in another pr possibly)

Reply:
I don't see an immediate use for this function but if you like we can also include both `_jpeg_encode` and `_jpeg_decode`.

Reply:
+1